include/uuid/uuid.h

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2023 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef _UUID_H_
+#define _UUID_H_
+
+namespace uprotocol::uuid {
+/*
+* Data model representation of <b>UUID</b>
+* UUID (Universal Unique Identifier) is a 128-bit value used to uniquely
+* identify an object or entity on the internet.
+* UUID(in here) represented as two long integers(64-bit) MSB and LSB(64 bit)
+* Class is designed as immutable
+*/
+class UUID {
+public:
+    /**
+    * Constructor to build UUID object - from msb and lsb
+    */
+    UUID(uint64_t msb,
```

Comment:
no need to pass msb and lsb to the constructor , all of this logic should be encapsulated

Reply:
> no need to pass msb and lsb to the constructor , all of this logic should be encapsulated

How to return UUID object during serialization classes, when msb and lsb are available?

Reply:
Discussed with Michael and Steven. Resolved.